proof/refine/ARM/IpcCancel_R.thy

```diff
@@ -1044,15 +1019,34 @@ lemma cancelSignal_tcb_obj_at':
   apply (wpsimp wp: setThreadState_not_st getNotification_wp)
   done
 
-lemma (in delete_one_conc_pre) cancelIPC_tcbDomain_obj_at':
+crunches replyRemoveTCB, cancelSignal, getBlockingObject
+  for obj_at'_only_st_qd_ft: "\<lambda>s. P (obj_at' (Q :: tcb \<Rightarrow> bool) t s)"
```

Comment:
Considering the name you've given to the following lemma, is his wha you'd like to call just this predicate?

Reply:
I'm not sure what you're asking here?

Reply:
You have `cancelIPC_obj_at'_only_st_qd_ft` just below this. That is for a predicate that doesn't care about state, queued, or fault updates. But here, you're crunching, and calling something `obj_at'_only_st_qd_ft`, but this predicate here is more general

Reply:
Right; this is making use of a `crunches` feature where it goes looking for lemmas named `${constant_name}_${property_name}`, letting those lemmas add side-conditions (I think I have a comment about this somewhere else in the PR). In short, this line doesn't produce a more general lemma, but that fact isn't visible/obvious, which is unfortunate.

Reply:
To be pedantic, `crunch` adds side-conditions from any lemmas that it finds and reuses, not just those with the appropriate name. It can also find lemmas that it has already proven in the current command or from the `wp` set.

Reply:
So is the odd naming here actually beneficial?